### PR TITLE
fix: render props rework

### PR DESCRIPTION
**Pull request by havenchyk**
### Description

Replace customRender functions with renderProps variants for everything except Calendar

### Review

- [x] Verify that all classes affected by the changes have **good** top-level documentation.
- [ ] Verify that new code has been covered with specs (and features if applicable).
- [ ] Verify that a corresponding meaningful changelog entry has been added (`* Add Github pull request template`), for this change if applicable.
- [ ] Test manually.
- [ ] Get two :+1: from code review.

### Pre-merge checklist

- [ ] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [ ] Verify that all new dependencies are included in `package.json`.
- [ ] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [ ] Squash related commits together.
- [ ] Verify the branch name starts with `NNNN` where `NNNN` is the issue number.

### CODE REVIEW

**Comment by havenchyk** [comment]
Well, I thought it would be too much for one PR, there is a huge PR from Peter which we never merged

File: src/RenderPropsComponents/HeaderButton/test.tsx [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+import {
+  BLOCK_CLASS_NAME,
+  NEXT_MONTH_TITLE,
+  PREV_MONTH_TITLE
+} from '../../calendar/consts'
+import HeaderButton from '.'
 
 describe('HeaderButton', () => {
-  let props, wrapper
+  let props: any, wrapper: any
```
Comment (PeterMK85) [inline src/RenderPropsComponents/HeaderButton/test.tsx:12]: `props` -> ComponentProps<typeof HeaderButton>

Comment (havenchyk) [reply src/RenderPropsComponents/HeaderButton/test.tsx:12]: for now it should be fine 😄

**Comment by havenchyk** [comment]
🤞

**Pull request merged**